Implement DBus try-commit handling

slyon opened the pull request:
## Description
Implement the `Try(uint32 timeoutInSeconds)` and `Cancel()` command inside `netplan-dbus` and adopt the `Apply()` command according to spec. "Try(...)" calls `netplan try`, keeping `stdin` open. The "Apply()" command uses the `SIGUSR1` signal, to send a confirmation to the `netplan try` process to confirm/commit the settings. The "Cancel()" command uses the `SIGINT` signal, to send a rejection to the `netplan try` child process.
The child process is being monitored, using events on the sd-event mainloop.

Codecov reports a reduced coverage, but that is only because the code in `terminal.py` cannot be tested from a script, as it uses the `isatty()` condition. And that part is excluded/skipped from our test-suite if run inside a script. If `make check-coverage` is executed locally, it retains 100% coverage.

## Checklist

- [x] Runs `make check` successfully.
- [x] Retains 100% code coverage (`make check-coverage`).
- [ ] New/changed keys in YAML format are documented.
- [ ] \(Optional\) Closes an open bug in Launchpad.

sil2100 commented:
On it, but I might need a bit to properly review!

sil2100 commented on src/dbus.c:
```diff
@@ -183,24 +332,23 @@ int main(int argc, char *argv[]) {
+    if (r < 0)
+        fprintf(stderr, "Failed mainloop: %s\n", strerror(-r));
 finish:
+    g_free(data);
```
Do we also need/should do a `sd_event_unref(event);` here? Or maybe it's not really necessary?

slyon replied:
Absolutely! This is the proper way to clean it up. The application terminates afterwards, so it doesn't make that big of a difference, but still we should have it properly cleaned up in the code.

Fixed.

sil2100 commented on src/dbus.c:
```diff
@@ -18,11 +20,86 @@
+    /* Accept the current 'netplan try', if active.
+     * Otherwise execute 'netplan apply' directly. */
+    if (d->try_pid > 0)
+        return _try_accept(TRUE, m, userdata, ret_error);
```
Oh, good catch with this case here!

sil2100 reviewed (approved):
I must say that I'm not a DBus expert, so this took longer for me to process. But architecture-wise and code-wise this looks like a good hit. Will still play around with it locally myself later today, but it's all good as is. +1

slyon commented:
Thank you for the review!

I've fixed the `sd_event` cleanup and removed some unneeded safeguards, for easier testability.

Now the code can be executed locally via:
`sudo LD_LIBRARY_PATH=/home/lukas/canonical/netplan PYTHONPATH=/home/lukas/canonical/netplan DBUS_TEST_NETPLAN_CMD=/home/lukas/canonical/netplan/src/netplan.script ./netplan-dbus`

And called via DBus:
`sudo busctl call io.netplan.Netplan /io/netplan/Netplan io.netplan.Netplan Try u 20` (or `Apply`, `Cancel`, ...)
